### PR TITLE
clib: Improve docstrings, comments, type hints and codes of some Session methods

### PR DESCRIPTION
**Pull request by seisman**
Following methods are updated minorly:

- `Session.get_common`
- `Session.extract_region`
- `Session.virtualfile_to_raster`
- `Session.virtualfile_to_dataset`
- `Session.virtualfile_out`

**Preview**: https://pygmt-dev--3327.org.readthedocs.build/en/3327/api/index.html

### CODE REVIEW

File: pygmt/clib/session.py [hunk 1]
```diff
@@ -1803,6 +1802,12 @@ def read_virtualfile(
             Cast the data into a GMT data container. Valid values are ``"dataset"``,
             ``"grid"`` and ``None``. If ``None``, will return a ctypes void pointer.
 
+        Returns
+        -------
+        pointer
+            Pointer to the GMT data container. If ``kind`` is None, returns a ctypes
+            void pointer instead.
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:1809]:
```suggestion
        pointer
            Pointer to the GMT data container. If ``kind`` is ``None``, returns a ctypes
            void pointer instead.
```

File: pygmt/clib/session.py [hunk 2]
```diff
@@ -2068,54 +2061,49 @@ def virtualfile_to_raster(
...
+        ...     region = lib.extract_region()
+        >>> print(", ".join([f"{x:.2f}" for x in region]))
         -164.71, -154.81, 18.91, 23.58
 
-        The country codes can have an extra argument that rounds the region a
-        multiple of the argument (for example, ``"US.HI+r5"`` will round the
-        region to multiples of 5):
+        The country codes can have an extra argument that rounds the region a multiple
+        of the argument (for example, ``"US.HI+r5"`` will round the region to multiples
+        of 5):
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:2100]:
```suggestion
        The country codes can have an extra argument that rounds the region to a multiple
        of the argument (for example, ``"US.HI+r5"`` will round the region to multiples
        of 5):
```

File: pygmt/clib/session.py [hunk 3]
```diff
@@ -2068,54 +2061,49 @@ def virtualfile_to_raster(
...
-
         >>> import pygmt
         >>> fig = pygmt.Figure()
         >>> fig.coast(
-        ...     region=[0, 10, -20, -10],
-        ...     projection="M6i",
-        ...     frame=True,
-        ...     land="black",
+        ...     region=[0, 10, -20, -10], projection="M12c", frame=True, land="black"
         ... )
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:2083]:
```suggestion
        ...     region=[0, 10, -20, -10], projection="M12c", frame=True, land="black",
        ... )
```

Comment (seisman) [reply pygmt/clib/session.py:2083]: I don't think `ruff` likes this trailing comma.

Comment (yvonnefroehlich) [reply pygmt/clib/session.py:2083]: OK. I was wondering, if this comma is required. But if not simply ignore this review.

**Review by michaelgrund — APPROVED** [review]

**Review by yvonnefroehlich — COMMENTED** [review]
Please fix the typo in line 377 / 378 ("fist" to "first").

**Pull request merged**